test/priceUtils.js

```diff
@@ -26,5 +33,123 @@ contract("PriceOracle", function(accounts) {
+      globalPriceStorage["WETH-USDC"] = 1 / 120.0
+
+      assert.equal(await checkNoProfitableOffer(orders[0], exchange, globalPriceStorage), true)
+      assert.equal(await checkNoProfitableOffer(orders[1], exchange, globalPriceStorage), true)
+    })
+    it("checks that bracket traders does not sell unprofitable for tokens with the different decimals", async () => {
+      const ERC20 = artifacts.require("DetailedMintableToken")
+      const token1 = await ERC20.new("DAI", 18)
+      const token2 = await ERC20.new("USDC", 6)
+
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token1.address, { from: accounts[0] })
+      const DAItokenId = (await exchange.numTokens.call()).toNumber() - 1
+
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token2.address, { from: accounts[0] })
+      const USDCtokenId = (await exchange.numTokens.call()).toNumber() - 1
+
+      const orders = [
+        {
+          // normal order selling for more than 1
+          user: "0x4c7281e2bd549a0aea492b28ef60e3d81fed36e6",
+          sellTokenBalance: new BN("10eed9efc"),
+          buyToken: DAItokenId,
+          sellToken: USDCtokenId,
+          priceNumerator: new BN("101").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("100").mul(new BN(10).pow(new BN(6))),
+        },
+        {
+          // normal order selling for more than 1
+          user: "0x4c7281e2bd549a0aea492b28ef60e3d81fed36e6",
+          sellTokenBalance: new BN("0"),
+          buyToken: USDCtokenId,
+          sellToken: DAItokenId,
+          priceNumerator: new BN("101").mul(new BN(10).pow(new BN(6))),
+          priceDenominator: new BN("100").mul(new BN(10).pow(new BN(18))),
+        },
+      ]
+
+      const globalPriceStorage = {}
+      globalPriceStorage["USDC-USDC"] = 1.0
+      globalPriceStorage["DAI-USDC"] = 1.0
+      assert.equal(await checkNoProfitableOffer(orders[0], exchange, globalPriceStorage), true)
+      assert.equal(await checkNoProfitableOffer(orders[1], exchange, globalPriceStorage), true)
+    })
+    it("detects unprofitable orders for tokens with different decimals", async () => {
+      const ERC20 = artifacts.require("DetailedMintableToken")
+      const token1 = await ERC20.new("DAI", 18)
+      const token2 = await ERC20.new("USDC", 6)
+
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token1.address, { from: accounts[0] })
+      const DAItokenId = (await exchange.numTokens.call()).toNumber() - 1
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token2.address, { from: accounts[0] })
+      const USDCtokenId = (await exchange.numTokens.call()).toNumber() - 1
+
+      const orders = [
+        {
+          // order is profitable for others
+          user: "0x4c7281e2bd549a0aea492b28ef60e3d81fed36e6",
+          sellTokenBalance: new BN("10eed9efc"),
```

Comment:
This should definitely fail, since `10eed9efc` is not a number. The standard way to get a value from a hex string is `new BN("10eed9efc", 16)`. This issue was fixed in PR #111, so it's probably better to merge master into this PR to get the last updates.

Reply:
Just out of curiosity, where is this number from?

Reply:
I copied real orders